pkg/scheduler/util/utils.go

```diff
@@ -188,7 +188,7 @@ func GetPodNonPreempting(pod *v1.Pod, kubeClient clientset.Interface) bool {
 	if kubeClient == nil {
 		return false
 	}
-	priorityClass, _ := kubeClient.SchedulingV1beta1().PriorityClasses().Get(pod.Name, metav1.GetOptions{})
+	priorityClass, _ := kubeClient.SchedulingV1beta1().PriorityClasses().Get(pod.Spec.PriorityClassName, metav1.GetOptions{})
```

Comment:
I'd really prefer having a cache here that gets updated using `informers`. But I'll defer to @bsalamat and @k82cn

Reply:
Yes, getting PriorityClasses from the API server in critical path of scheduling could have major performance implications. We should use informers and cache them.